src/app/InteractionModelDelegate.h

```diff
@@ -196,13 +197,24 @@ class InteractionModelDelegate
     }
 
     /**
-     * Notification that a read client has completed the read interaction.
+     * Notification that a Subscribe Response has been processed and application can do further work .
+     */
+    virtual CHIP_ERROR SubscribeResponseProcessed(const ReadClient * apReadClient) { return CHIP_ERROR_NOT_IMPLEMENTED; }
+
+    //TODO: Refactor callback delegate into client and server delegate for different IM interaction
+    /**
+     * Notification that Subscription has been established successfully and application can do further work in handler.
+     */
+    virtual CHIP_ERROR SubscriptionEstablished(const ReadHandler * apReadHandler) { return CHIP_ERROR_NOT_IMPLEMENTED; }
+
+    /**
+     * Notification that a read client has completed the current interaction.
```

Comment:
This is specific to a read, and not a subscription right? Because for a subscription, it lives on indefinitely till either the client or server terminate the subscription.

Comment should be updated to indicate that this is only for _successful_ reads:

```suggestion
     * Notification that a read interaction was completed on the client successfully.
```

Reply:
Done, thanks